modulefiles/obsproc_hera.lua

```diff
@@ -2,12 +2,15 @@ help([[
 Load environment to build obsproc on Hera
 ]])
 
-load("cmake/3.20.1")
+prepend_path("MODULEPATH", pathJoin("/scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-" .. (os.getenv("spack_stack_ver") or "None"), "/envs/" .. (os.getenv("spack_env") or "None"), "-dev-rocky8/install/modulefiles/Core"))
 
-prepend_path("MODULEPATH", "/scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/modulefiles/stack")
-load("hpc/1.1.0")
-load("hpc-intel/18.0.5.274")
-load("hpc-impi/2018.0.4")
+stack_intel_ver=os.getenv("stack_intel_ver") or "2021.5.0"
+stack_impi_ver=os.getenv("stack_impi_ver") or "2021.5.1"
+cmake_ver=os.getenv("cmake_ver") or "3.23.1"
```

Comment:
I suggest we remove the defaults here; i.e. the `or` part for spack installed stuff.  They are either defined in `build.spack.ver` or in `build.platform.ver`. That way, these modulefiles will likely only see updates to the `MODULEPATH`, which I think we should also define in `build.platform.ver` to something like `spack_modpath=....`. Take away all the gory details and clump them in `build.platform.ver` and `run.platform.ver`.
These module files then are free of any specifics.  Thoughts?

Reply:
Agreed. Removed defaults @ dcd0a43.